docs/UA-service-API.md

```diff
@@ -0,0 +1,456 @@
+response body will depend on the endpoint in question.
+
+Failures due to invalid behavior from the client will produce a
+response with HTTP status code in the "4XX" range and content-type of
+"application/json".  Failures due to an unexpected situation on the
+server will produce a response with HTTP status code in the "5XX"
+range and content-type of "application/json".
+
+---
+
+TODO: diffs WebSocket API documentation
+
+---
+
+# API Endpoints
+
+* Sessions
+    * [POST /v1/session/start](#post-v1sessionstart)
+    * [POST /v1/session/end](#get-v1sessionend)
+
+* History visits
+    * [POST /v1/visits](#post-v1visits)
+    * [GET /v1/visits](#get-v1visits)
+
+* Page details
+    * [POST /v1/pages/:url](#post-v1pagesurl)
+    * [GET /v1/query](#get-v1query)
+
+* Stars
+    * [PUT /v1/stars/:url](#put-v1starsurl)
+    * [DELETE /v1/stars/:url](#delete-v1starsurl)
+    * [GET /v1/stars](#get-v1stars)
+    * [GET /v1/recentStars](#get-v1recentstars)
+
+## POST /v1/session/start
+
+Start a new browsing session.  A session roughly corresponds to a tab.
+
+___Body parameters___
+
+* scope - (optional) integer session ID to mark as the scope of this session.
+* ancestor - (optional) integer session ID to mark as the ancestor of this session.
+
+### Request
+
+```sh
+curl -v \
+-X POST \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/session/start" \
+-d '{
+  "scope": 0,
+  "session: 100
+}'
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with the
+allocated session ID in the JSON body:
+
+```json
+{
+  "session": 101
+}
+```
+
+## POST /v1/session/end
+
+End an existing browsing session.  A session roughly corresponds to a tab.
+
+___Body parameters___
+
+* session - integer session ID to end.
+
+### Request
+
+```sh
+curl -v \
+-X POST \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/session/end \
+-d '{
+  "session: 100
+}'
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with an empty JSON body:
+
+```json
+{
+}
+```
+
+## POST /v1/visits
+
+Record a visit to a URL, optionally adding a title event to that URL.
+
+___Body parameters___
+
+* session - integer session ID to associate visit with.
+* url - string URL to visit.
+* title (optional) - string title of visited URL.
+
+### Request
+
+```sh
+curl -v \
+-X POST \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/session/end \
+-d '{
+  "session: 100,
+  "url": "https://www.mozilla.org/en-US/firefox/new/",
+  "title": "Download Firefox - Free Web Browser",
+}'
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with an empty JSON body:
+
+```json
+{
+}
+```
+
+## GET /v1/visits
+
+Return visits to all URLs, optionally limiting the number of visits
+returned.  See also [GET /v1/query](#get-v1query).
+
+___Query parameters___
+
+* limit - (optional) integer maximum number of visits to return.
```

Comment:
Is this actually optional?  Looks like the validator is expecting it to not be empty: https://github.com/mozilla/tofino/blob/master/app/services/user-agent-service/server.js#L128